From f497eef1f38bd79a96a13269f251f1413c54b790 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Tue, 25 Mar 2025 12:36:32 +0100
Subject: [PATCH] Fix build with gcc-15

* fixes:
  http://errors.yoctoproject.org/Errors/Details/848747/

usb_modeswitch.c: In function 'main':
usb_modeswitch.c:573:28: error: too many arguments to function 'get_current_config_value'; expected 0, have 1
  573 |         currentConfigVal = get_current_config_value(dev);
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~ ~~~
In file included from usb_modeswitch.c:59:
usb_modeswitch.h:55:5: note: declared here
   55 | int get_current_config_value();
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
usb_modeswitch.c:775:52: error: too many arguments to function 'get_current_config_value'; expected 0, have 1
  775 |                                 currentConfigVal = get_current_config_value(dev);
      |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~ ~~~
usb_modeswitch.h:55:5: note: declared here
   55 | int get_current_config_value();
      |     ^~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
Upstream-Status: Submitted [https://www.draisberghof.de/usb_modeswitch/bb/viewtopic.php?t=3122]
---
 usb_modeswitch.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usb_modeswitch.c b/usb_modeswitch.c
index 2d4c0aa..bfbcbfb 100644
--- a/usb_modeswitch.c
+++ b/usb_modeswitch.c
@@ -570,7 +570,7 @@ int main(int argc, char **argv)
 	/* Get current configuration of default device, note value if Configuration
 	 * parameter is set. Also sets active_config
 	 */
-	currentConfigVal = get_current_config_value(dev);
+	currentConfigVal = get_current_config_value();
 	if (Configuration > -1) {
 		SHOW_PROGRESS(output,"Current configuration number is %d\n", currentConfigVal);
 	} else
@@ -772,7 +772,7 @@ int main(int argc, char **argv)
 	if (Configuration > 0) {
 		if (currentConfigVal != Configuration) {
 			if (switchConfiguration()) {
-				currentConfigVal = get_current_config_value(dev);
+				currentConfigVal = get_current_config_value();
 				if (currentConfigVal == Configuration) {
 					SHOW_PROGRESS(output,"The configuration was set successfully\n");
 				} else {
